patches/tModLoader/Terraria/Item.TML.cs

```diff
@@ -28,6 +28,15 @@ public partial class Item : TagSerializable, IEntityWithGlobals<GlobalItem>
 			set => _damageClass = value ?? throw new ArgumentException("DamageType cannot be null");
 		}
 
+		private int _armorPenetration = 0;
+		/// <summary>
+		/// The number of defense points that this item can ignore on its own. Cannot be set to negative values. Defaults to 0.
+		/// </summary>
+		public int ArmorPenetration {
+			get => _armorPenetration;
+			set => _armorPenetration = Math.Max(value, 0);
```

Comment:
I'm concerned this may cause confusion when some mod inevitably sets it to a negative value and successfully builds and runs it without testing this in particular.
Consider tossing an exception instead so the modder is obligated to fix the stupid on their end.